### PR TITLE
Alerting: Fix recording rules showing alert notification information.

### LINKED ISSUE #61495: Alert notifications information is shown when creating a Mimir or Loki managed recording rule
**What happened**:
When I go to create a recording rule in the Alerting UI, I'm shown a prompt about configuring Notifications and Alerting routes. See attached screenshots.

**What you expected to happen**:
I should not see this because that information is relevant to alerting rules, but not recording rules. 

alerting-bug2
alerting-bug1

**Environment**:
- Grafana version: Grafana Cloud Pro [v9.3.2-67a213dc85 (67a213dc85)](https://github.com/grafana/grafana/blob/main/CHANGELOG.md)
- User OS & Browser: Chrome

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR fixes recording rules form showing information related to alerting (notifications, preview alert button, ...)
This PR also fix docs regarding the creation of cloud and recording rules.

**Why do we need this feature?**

Recording rules are not used to generate alert instances, and for this reason they should not show information regarding notifications and alerts firing.

**Who is this feature for?**

All users.

**Which issue(s) does this PR fix?**:

Fixes #61495 

**Special notes for your reviewer**:

https://user-images.githubusercontent.com/33540275/214292544-2e0ec1c9-6de2-441e-a17a-d47e0182f23d.mp4

### CODE REVIEW

**Review by VikaCep — APPROVED** [review]
💯

File: docs/sources/alerting/alerting-rules/create-mimir-loki-managed-recording-rule.md [hunk 1]
```diff
@@ -36,19 +36,17 @@ To create a Grafana Mimir or Loki managed recording rule
 
 1. In the Grafana menu, click the **Alerting** (bell) icon to open the Alerting page listing existing alerts.
 1. Click **New alert rule**. The new alerting rule page opens where the **Grafana managed alert** option is selected by default.
-1. In Step 1, add the rule name.
-   - In **Rule name**, add a descriptive name. This name is displayed in the alert rule list. It is also the `alertname` label for every alert instance that is created from this rule.
+1. In Step 1, add the rule name. Recording rule name must be valid metric name.
```
Comment (brendamuir) [inline docs/sources/alerting/alerting-rules/create-mimir-loki-managed-recording-rule.md:39]:
```suggestion
1. In Step 1, add the rule name. The recording rule name must be a valid metric name.
```

Comment (brendamuir) [reply docs/sources/alerting/alerting-rules/create-mimir-loki-managed-recording-rule.md:39]: What does "must be a valid metric name" mean? could we maybe write this more clearly?

File: docs/sources/alerting/alerting-rules/create-mimir-loki-managed-recording-rule.md [hunk 2]
```diff
@@ -36,19 +36,17 @@ To create a Grafana Mimir or Loki managed recording rule
...
-   - From the **Group** drop-down, select an existing group within the selected namespace. Otherwise, click **Add new** and enter a name to create a new one. Newly created rules are appended to the end of the group. Rules within a group are run sequentially at a regular interval, with the same evaluation time.
-   - Add a description and summary to customize alert messages. Use the guidelines in [Annotations and labels for alerting]({{< relref "../fundamentals/annotation-label/" >}}).
-   - Add Runbook URL, panel, dashboard, and alert IDs.
-   - Add custom labels.
-1. Click **Save** to save the rule or **Save and exit** to save the rule and go back to the Alerting page.
-1. Next, create a notification for the rule.
+   - From the **Group** drop-down, select an existing group within the selected namespace. Otherwise, click **Add new** and enter a name to create a new one.
+1. In Step 4, add the custom labels.
+   - Add custom labels selecting existing key-value pairs from the drop down, or add new labels by entering the new key or value .
+1. Click **Save** to save the recording rule or **Save and exit** to save the recording rule and go back to the Alerting page.
```
Comment (brendamuir) [inline docs/sources/alerting/alerting-rules/create-mimir-loki-managed-recording-rule.md:49]:
```suggestion
   - Add custom labels selecting existing key-value pairs from the drop down, or add new labels by entering the new key or value.
1. Click **Save** to save the recording rule or **Save and exit** to save the recording rule and go back to the Alerting page.
```

File: docs/sources/alerting/alerting-rules/create-mimir-loki-managed-recording-rule.md [hunk 3]
```diff
@@ -57,13 +55,12 @@ To create a Grafana Mimir or Loki managed recording rule
...
-   - From the **Namespace** drop-down, select an existing rule namespace. Otherwise, click **Add new** and enter a name to create a new one. Namespaces can contain one or more rule groups and only have an organizational purpose.
-   - From the **Group** drop-down, select an existing group within the selected namespace. Otherwise, click **Add new** and enter a name to create a new one. Newly created rules are appended to the end of the group. Rules within a group are run sequentially at a regular interval, with the same evaluation time.
-     {{< figure src="/static/img/docs/alerting/unified/rule-edit-mimir-alert-type-8-0.png" max-width="550px" caption="Alert details" >}}
-1. In Step 3, add the query to evaluate.
    - Enter a PromQL or LogQL expression. The rule fires if the evaluation result has at least one series with a value that is greater than 0. An alert is created for each series.
-     {{< figure src="/static/img/docs/alerting/unified/rule-edit-mimir-query-8-0.png" max-width="550px" caption="Alert details" >}}
+1. In Step 3, add evaluation behaviour.
+   - Enter a valid **For** duration. Expression has to be true for this long for the alert to be fired.
```
Comment (brendamuir) [inline docs/sources/alerting/alerting-rules/create-mimir-loki-managed-recording-rule.md:60]:
```suggestion
1. In Step 3, add evaluation behavior.
   - Enter a valid **For** duration. The expression has to be true for this long for the alert to be fired.
```

File: docs/sources/alerting/alerting-rules/create-mimir-loki-managed-recording-rule.md (hunk 1 above)
Comment (soniaAguilarPeiron) [reply docs/sources/alerting/alerting-rules/create-mimir-loki-managed-recording-rule.md:39]: means that Recording rule names must also be valid prometheus metric names, that is, contain no whitespace. Maybe we could add this requirement 👍

File: docs/sources/alerting/alerting-rules/create-mimir-loki-managed-recording-rule.md [hunk 4]
```diff
@@ -36,19 +36,17 @@ To create a Grafana Mimir or Loki managed recording rule
 
 1. In the Grafana menu, click the **Alerting** (bell) icon to open the Alerting page listing existing alerts.
 1. Click **New alert rule**. The new alerting rule page opens where the **Grafana managed alert** option is selected by default.
-1. In Step 1, add the rule name.
-   - In **Rule name**, add a descriptive name. This name is displayed in the alert rule list. It is also the `alertname` label for every alert instance that is created from this rule.
+1. In Step 1, add the rule name. The recording name must also be Prometheus metric name and contain no whitespace.
```
Comment (brendamuir) [inline docs/sources/alerting/alerting-rules/create-mimir-loki-managed-recording-rule.md:39]:
```suggestion
1. In Step 1, add the rule name. The recording name must be a Prometheus metric name and contain no whitespace.
```

**Review by brendamuir — APPROVED** [review]

**Pull request merged**